src/content/blog/2022-08-03-new-config-system-part-1.md

```diff
@@ -0,0 +1,130 @@
+
+This also turned out, in hindsight, to not be a great idea. Adding a JavaScript config file format created an incompatibility between it and the non-JS formats: any JavaScript object could be passed into the config and available in rules. Because we didn't properly validate the config to exactly match the non-JS formats, we ended up with some rules requiring regular expression objects to be passed in to be properly configured. While this could work in the JS config file format, the rules could not be properly configured in non-JS config files. Unfortunately, because plugin rules depended on this functionality, we couldn't go back and fix it without breaking things.
+
+### Shareable configs and dependencies
+
+Perhaps the biggest problem we faced early on was when npm decided to [stop installing peer dependencies in v3](https://github.com/npm/npm/issues/6565#issuecomment-74971689). Prior to this point, we had recommended that shareable configs include any plugins they depended on as peer dependencies rather than regular dependencies. This was a quirk of the way that `extends` was implemented: using `require()`.
+
+Because shareable configs were data-only and couldn't directly reference Node.js dependencies, `require()` would not automatically load direct dependencies into the path for ESLint to resolve them. Peer dependencies, on the other hand, worked perfectly by just using `require()` because those were installed in a location where the normal package lookup worked.
+
+When npm v3 stopped installing peer dependencies by default, all of the shared configs relying on this behavior stopped working correctly. There is a [long-running issue](https://github.com/eslint/eslint/issues/3458) requesting that shareable configs be allowed to use dependencies directly, but the architecture of eslintrc just didn't allow for it. We would have essentially had to recreate the entire `require()` functionality inside of ESLint to work around the way shareable configs were designed. We recommended that shareable configs create a post-install script to install their peer dependencies instead. Not ideal by any stretch of the imagination.
+
+We added the `--resolve-plugins-relative-to` command line option to try and help with this problem, but it wasn't enough. The most popular requests for help in our [Discord #help channel](https://eslint.org/chat/help) have to do with improper resolution of plugins from config files.
+
+npm eventually changed back to installing peer dependencies by default in v7, but by that point the damage on the ESLint ecosystem has been done.
```

Comment:
and some other package managers(e.g. yarn/pnpm) just won't install peer dependencies by default - same as npm v3~v6.